hydromt/models/components/mesh.py

```diff
@@ -10,34 +10,389 @@
 import xarray as xr
 import xugrid as xu
 from pyproj import CRS
+from shapely.geometry import box
 
-from hydromt import workflows
 from hydromt.gis.raster import GEO_MAP_COORD
+from hydromt.io.readers import read_nc
 from hydromt.models.api import Model
+from hydromt.models.components.base import ModelComponent
+from hydromt.workflows.mesh import (
+    create_mesh2d,
+    mesh2d_from_raster_reclass,
+    mesh2d_from_rasterdataset,
+    rename_mesh,
+)
 
-__all__ = ["MeshModel"]
-logger = logging.getLogger(__name__)
+logger = getLogger(__name__)
 
 
-class MeshMixin(object):
-    # placeholders
-    # We cannot initialize an empty xu.UgridDataArray
-    _API = {
-        "mesh": Union[xu.UgridDataArray, xu.UgridDataset],
-    }
+class MeshComponent(ModelComponent):
+    """ModelComponent class for mesh components."""
 
-    def __init__(self, *args, **kwargs) -> None:
-        super().__init__(*args, **kwargs)
-        self._mesh = None
+    def __init__(self, model: Model):
+        super().__init__(model=model)
+        self._data = None
 
-    ## general setup methods
-    def setup_mesh2d_from_rasterdataset(
+    def set(
+        self,
+        data: Union[xu.UgridDataArray, xu.UgridDataset],
+        name: Optional[str] = None,
+        grid_name: Optional[str] = None,
+        overwrite_grid: bool = False,
+    ) -> None:
+        """Add data to mesh.
+
+        All layers of mesh have identical spatial coordinates in Ugrid conventions.
+        Also updates self.region if grid_name is new or overwrite_grid is True.
+
+        Parameters
+        ----------
+        data: xugrid.UgridDataArray or xugrid.UgridDataset
+            new layer to add to mesh, should contain only one grid topology.
+        name: str, optional
+            Name of new object layer, this is used to overwrite the name of
+            a UgridDataArray.
+        grid_name: str, optional
+            Name of the mesh grid to add data to. If None, inferred from data.
+            Can be used for renaming the grid.
+        overwrite_grid: bool, optional
+            If True, overwrite the grid with the same name as the grid in self.mesh.
+        """
+        # Checks on data
```

Comment:
I expected the first call to be `_initialize_mesh()`, just like other components.

Reply:
I think it is done this way because you cannot create an empty UgridDataset

Reply:
Indeed! So the choice would either be xr.Dataset or None but I think None is easier and clearer to handle.